packages/components/src/higher-order/with-focus-return/test/index.js

```diff
@@ -108,50 +108,10 @@ describe( 'withFocusReturn()', () => {
 
 			// Should return to the activeElement saved with this component.
 			unmount();
+			render( <div></div>, {
+				container,
+			} );
 			expect( document.activeElement ).toBe( activeElement );
 		} );
-
-		it( 'should switch focus to the most recent still-available focus target', () => {
```

Comment:
The previous implementation supported that use-case but IMO it was just something implemented to "hide" other bugs we had about focus handling. The recent refactors (removing auto-focus) make this not useful anymore. I also decided to remove it because just to support this, we need extra providers and a context which are not needed anymore (so more portable)

Reply:
It might be something with autofocus but it might be something related to sidebars as well. I can check the history as I probably helped to land the original PR. One use case that we can test manually is to open Options from More Menu and check whether focus returns to the menu item.

Reply:
Another use case that I see now that regressed is the document/block setting sidebar. In the past, the focus would be moved to the toggle button when closed but now it gets moved to the `body` element 🤷🏻‍♂️

Reply:
I can confirm now that I found regression:

![Dec-09-2020 20-06-00](https://user-images.githubusercontent.com/699132/101675387-04fba380-3a5a-11eb-885f-8197292792ae.gif)

Reply:
I believe the issue here might be something in the Modal component way of handling focusOnMount.  I'll debug tomorrow. If we all consolidate on the different hooks I'm adding to the compose package, these things will get easier, simpler and more consistent.

Reply:
My idea here is that useFocusOnMount and useFocusReturn are implementing the same behavior in two different directions so using them together should be the way to go ultimately. I'll do that with a usePopover or useDialog hook but for the moment, I'll see if I can fix the modal issue with a simpler fix.

Reply:
Whatever works for you best, it's a complex behavior so I would be happy to come up with a more general fix because what we had before was very hacky and it depended on the role `modal` 😄

Reply:
This should be fixed now.

Reply:
It works, awesome 👏🏻